aiida/engine/processes/calcjobs/calcjob.py

```diff
@@ -516,8 +535,20 @@ def parse(self, retrieved_temporary_folder: Optional[str] = None) -> ExitCode:
         for entry in self.node.base.links.get_outgoing():
             self.out(entry.link_label, entry.node)
 
+        if existing_exit_code is not None:
+            return existing_exit_code
+
         return exit_code or ExitCode(0)
 
+    @staticmethod
+    def terminate(exit_code: ExitCode) -> ExitCode:
+        """Terminate the process immediately and return the given exit code.
+
+        :param exit_code: The exit code to return.
+        :returns: The provided exit code.
+        """
+        return exit_code
```

Comment:
This is called by the `Waiting.execute` method if a monitor triggers and sets `parse=false`. Normally, the execute method recreates the `RUNNING` state with the `parse` method to be called, but here we need another method that will simply terminate the process. Any process step that returns an exit code will cause it to be terminated, so that is what this hook allows us to do. There was no other process step that was a no-op like this (I don't believe).

Reply:
I have updated the docstring to explain this.